### PR TITLE
Fix(GUI): Redesign and fix parts of the gui to be more responsive

### LINKED ISSUE #235: Improve design with Material 3
## User story
1. As a user
2. I want a nice UI
3. So that I can use the software easily

## Acceptance criteria
* upfront the meeting: 
* adjust colors that they are readable 
* text overflows in buttons should be fixed
* fix the displaying of the statistical information
* there should be a 2h meeting together with the PO's to finish the final UI 
* the UI should be reviewed by minimum 3 people

## Definition of done (DoD)
* Create feature branch
* Implement the functionality
* Opened pull request and assigned reviewer
* If necessary: required changes are implemented
* Delete feature branch
* If necessary: Component Test has been written
* Code review has been completed and code has been merged

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

### PR DESCRIPTION
**Pull request by simonsasse**
Closes #235 

### Diff view:
- [x] Fix save single image offset (Not perfectly fixed..)
- [x] Move buttons ins app bar from right to left (except help button)
- [x] Fix size of video frames (4:3 format looks weird)
- [x] Center frame titles above the frames
- [x] Move frame titles and buttons closer to frames
- [x] Reduce fullscreen button size (keep padding of icons in mind)
- [x] Center timelines horizontally in window
- [x] Refactor statistical informations - create info icon with title and tooltip (showing infos at hover) - position above the leftmost frame number (above the overview timeline)
### Fullscreen view
- [x] Fix size of frame to be actually full screen
- [x] Fix size of buttons to be same size as in diff screen
### Settings view:
- [x] Remove back button and implement app bar in settings view - move title to app bar
- [x] Reduce size of buttons (mask not whole width)
- [x] More tooltip offset (settings)
### Select view:
- [x] Fix filepath - bigger font size and earlier replacement with dots
### General:
- [x] Center titles horizontally in app bar (all screens)

#### Diff Screen
- [x] the full screen buttons scale too much, there should be a maximum size
- [x] the icon in the full screen buttons should scale linearly with button size. currently, there is a padding messing with that. It would be nice, if the proportions of the button and the icon always stay the same
- [x] The `Difference Screen` headline in the topbar overlaps with the buttons on the left, if the window's width is low enough
- [x] `Save image as png` does not pop up where the mouse was clicked (Well not really fixed)
#### Settings screen
- [x] I'd suggest putting the save settings button either in the bottom right or int the top bar on the right. Would be more streamlined with other programs' UI
#### Changes for other PRs
- Maskpath button not centered (but I guess we'll fix that with the merge of #255)
- Back buttons should ask for confirmation if data would be lost (can be done in a separate PR though)

### CODE REVIEW

**Comment by simonsasse** [comment]
Merge conflicts resolved.

**Comment by simonsasse** [comment]
Rebased 😓

**Comment by simonsasse** [comment]
This should be reviewed by 2-3 people before being merged !!

**Comment by akriese** [comment]
Great changes :)

Couple things, I noticed:
- [ ] the full screen buttons scale too much, there should be a maximum size
- [ ] the icon in the full screen buttons should scale linearly with button size. currently, there is a padding messing with that. It would be nice, if the proportions of the button and the icon always stay the same
- [ ] The `Difference Screen` headline in the topbar overlaps with the buttons on the left, if the window's width is low enough
- [ ] `Save image as png` does not pop up where the mouse was clicked
- [ ] Maskpath button not centered (but I guess we'll fix that with the merge of #255)

**Comment by zino212** [comment]
@akriese: For now I decided to just move the save button into the right corner.

Moving it into the appbar brings possible problems it, because the differentiation between saving and just going back would have to be made very clear. I don't know if positioning the button on the left next to the back button would be better or positioning it to the right. On the left the user would probably be more likely to see it than on the right but maybe it being so close to the back button would also be problematic.

In less words: The button being in the lower right corner is in my opinion for now the solution with the least problems. :)

**Comment by zino212** [comment]
The scaling of the full screen button seems to be a bit tricky. I set a maximum size for the vertical size for now. The horizontal size is responsive to the overall window size and therefore can not be set this way.
It would probably need a bigger refactoring to set that.

Additionally, I have no real clue on how to set a minimum size for the icons or how to make them scale linear with the buttons. I tried several things which didn't work out. Maybe @simonsasse has some ideas?
Regarding the overlapping title my ideas didn't work out as well. The borders of the different areas of the appbar seem to be kind of messed up. When scaling the window down, the title seems to notice the navigation area, but far to late. The save image button seems to be placed out of the regular navigations button area and therefore the title overlaps.

**Comment by simonsasse** [comment]
OKaaaay.
I made the appBar responsive to an extend. very narrow windows still overlap. But  I think for what's relevant, it's fine now. Buttons get collapsed into a dropdown menu.

I changed the fullscreen button to be a plain IconButton. Looks cleaner and does not scale ugly.

I centered the download dropdown on the image. I am sure, it is possible to get the exact location and place it there but maybe that's not the most relevant thing for now.

**Review by akriese — APPROVED** [review]
All in all, this LGTM. Let's wait for at least one more positive review though :)

File: GUI/src/main/kotlin/ui/components/diffScreen/Timeline.kt
Comment (simonsasse) [reply GUI/src/main/kotlin/ui/components/diffScreen/Timeline.kt:375]: Doesn't work, because drawLine is none composable and thus we cannot make calls to MaterialTheme.colorScheme..

**Pull request merged**
